plugins/experimental/ssl_session_reuse/src/ssl_utils.h

```diff
@@ -45,6 +47,35 @@ struct ssl_session_param {
   ~ssl_session_param();
 };
 
+class PluginThreads
+{
+public:
+  void
+  store(const pthread_t &th)
+  {
+    std::lock_guard<std::mutex> lock(threads_mutex);
+    threads_queue.push_back(th);
+  }
+
+  void
+  terminate()
+  {
+    std::lock_guard<std::mutex> lock(threads_mutex);
+    for (pthread_t th : threads_queue) {
+      ::pthread_cancel(th);
+    }
+    while (!threads_queue.empty()) {
+      pthread_t th = threads_queue.front();
+      ::pthread_join(th, nullptr);
+      threads_queue.pop_front();
+    }
+  }
+
+private:
+  std::deque<pthread_t> threads_queue;
+  mutable std::mutex threads_mutex;
```

Comment:
Mutable member data is only useful in a class that will have const instances.  This class has no public interface that would be accessible for a const instance.  When would there be a const instance of this object?

Reply:
I think you're right, removing mutable.